### ISSUE TITLE
oneOf models are rendered with incorrect member names [RustGenerator]

### ISSUE THREAD

**Issue by leigh-johnson** [asyncapi/modelina#1055]
#### Describe the bug
Given the following schema, member names and serde macros are rendered with the index of the member appended to the model name. 

```yaml
    WebrtcSettingsApplyRequest:
      name: WebrtcSettingsApplyRequest
      type: object
      additionalProperties: false
      properties:
        video_src:
          x-parser-schema-id: VideoSource
          oneOf:
            - $ref: "#/components/schemas/Camera"
            - $ref: "#/components/schemas/PlaybackVideo"
      required:
        - video_src
 ```

The rust code currently rendered:

```rust
// VideoSource represents a union of types: Camera0, PlaybackVideo1
#[derive(Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub enum VideoSource {
    #[serde(rename="Camera0")]
    Camera0(crate::Camera),
    #[serde(rename="PlaybackVideo1")]
    PlaybackVideo1(crate::PlaybackVideo),
}
```

#### Expected behavior

The schema above should render the following model:

```rust
// VideoSource represents a union of types: Camera, PlaybackVideo
#[derive(Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub enum VideoSource {
    #[serde(rename="Camera")]
    Camera(crate::Camera),
    #[serde(rename="PlaybackVideo")]
    PlaybackVideo(crate::PlaybackVideo),
}
```

**Comment by leigh-johnson** [comment]
Dropping some debugging notes. Looks like `ConstrainedUnionModel` is getting an array of `ConstrainedReferenceModel` with the wrong types (`Cat0`, `Dog1`, `StickInsect2`). 

![Screenshot from 2022-12-14 10-45-24](https://user-images.githubusercontent.com/2601819/207686177-ba49e2d7-1e22-4f31-86d7-4bd9781124be.png)

**Comment by leigh-johnson** [comment]
I just noticed `discriminator` isn't being handled correctly either, going to fix that in #1056 as well.
